core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -240,25 +239,156 @@ boolean checkRequestParameters(final RestRequest request, final RestChannel chan
     }
 
     void executeHandler(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
-        final RestHandler handler = getHandler(request);
+        // Request execution flags
+        boolean explicitMatchHandled = false;
+        boolean wildcardMatchHandled = false;
+
+        /*
+         * First try handlers mapped to explicit paths.
+         */
+        explicitMatchHandled = executeExplicitHandler(request, channel, client);
+
+        if (explicitMatchHandled == false) {
+            /*
+             * Fallback to handlers mapped to both explicit and wildcard paths.
+             */
+            wildcardMatchHandled = executeWildcardHandler(request, channel, client);
+        }
+
+        /*
+         * If request has not been handled, fallback to a bad request error.
+         */
+        if (explicitMatchHandled == false && wildcardMatchHandled == false) {
+            handleBadRequest(request, channel);
+        }
+    }
+    
+    private boolean executeExplicitHandler(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        /*
+         * Get the matching explicit handler (ie. ignore wildcard path matches)
+         * for this request, if one exists.
+         */
+        final RestHandler explicitHandler = getHandler(request, true);
+        if (explicitHandler != null) {
+            /*
+             * Handle valid REST request (the request matches an explicit
+             * handler).
+             */
+            explicitHandler.handleRequest(request, channel, client);
+            return true;
+        }
+
+        /*
+         * Get the map of matching explicit handlers (ie. ignore wildcard path
+         * matches) for this request, for the set of HTTP methods.
+         */
+        final HashSet<RestRequest.Method> explicitPathValidMethodSet = getValidHandlerMethodSet(request, true);
+        if (explicitPathValidMethodSet.size() > 0 
```

Comment:
I think that this block of code is identical to a block of code in `RestController#executeExplicitHandler`; can we collapse them into a method?